### PR TITLE
Application command localization

### PR DESCRIPTION
**Pull request by PerchunPak**
See #268 for more history, I need this functionality, and so I took the PR as it looks like unmaintained anymore.

### CODE REVIEW

File: lightbulb/decorators.py [hunk 1]
```diff
@@ -98,13 +98,17 @@ def command(
         inherit_checks (:obj:`bool`): Whether or not the command should inherit checks from the parent group. Only
             affects subcommands. Defaults to ``False``.
         pass_options (:obj:`bool`): Whether the command will have its options passed as keyword arguments when invoked.
+        name_localizations (Mapping[Union[hikari.locales.Locale, :obj:`str`], :obj:`s®tr`]): A set of name localizations for this command. Defaults to an empty dictionary.
```
Comment (davfsa) [inline lightbulb/decorators.py:101]:
```suggestion
        name_localizations (Mapping[Union[hikari.locales.Locale, :obj:`str`], :obj:`str`]): A set of name localizations for this command. Defaults to an empty dictionary.
```

File: lightbulb/commands/base.py [hunk 2]
```diff
@@ -172,6 +172,18 @@ class OptionLike:
     """
     autocomplete: bool = False
     """Whether the option should be autocompleted or not. This only affects slash commands."""
+    name_localizations: t.Mapping[t.Union[hikari.Locale, str], str] = dataclasses.field(default_factory=dict)
+    """
+    A set of name localizations for this option
+    
+    .. versionadded:: 2.2.5
```
Comment (davfsa) [inline lightbulb/commands/base.py:179]: Also, these versionadded would need a bump

File: lightbulb/decorators.py (hunk 1 above)
Comment (PerchunPak) [reply lightbulb/decorators.py]: Fixed in c5888c1b1d9e40f58b7d092bb3f5aa454968c7e7.

File: lightbulb/commands/base.py (hunk 2 above)
Comment (PerchunPak) [reply lightbulb/commands/base.py]: Fixed in c5888c1b1d9e40f58b7d092bb3f5aa454968c7e7.

File: lightbulb/commands/base.py [hunk 3]
```diff
@@ -172,6 +172,18 @@ class OptionLike:
...
     """Whether the option should be autocompleted or not. This only affects slash commands."""
+    name_localizations: t.Mapping[t.Union[hikari.Locale, str], str] = dataclasses.field(default_factory=dict)
+    """
+    A set of name localizations for this option
+
+    .. versionadded:: 2.3.0
+    """
+    description_localizations: t.Mapping[t.Union[hikari.Locale, str], str] = dataclasses.field(default_factory=dict)
+    """
+    A set of description localizations for this option
```
Comment (tandemdude) [inline lightbulb/commands/base.py:183]: These docstrings should be changed as it is incorrect. `"A mapping of locale to ... localizations for this option"` would be more appropriate

File: lightbulb/commands/base.py [hunk 4]
```diff
@@ -288,6 +302,18 @@ class CommandLike:
...
     """Whether invocations of this command will bypass author permission checks, if an application command."""
+    name_localizations: t.Mapping[t.Union[hikari.Locale, str], str] = dataclasses.field(default_factory=dict)
+    """
+    A set of name localizations for this command
+    
+    .. versionadded:: 2.3.0
+    """
+    description_localizations: t.Mapping[t.Union[hikari.Locale, str], str] = dataclasses.field(default_factory=dict)
+    """
+    A set of description localizations for this command
```
Comment (tandemdude) [inline lightbulb/commands/base.py:313]: Same as above. `"A mapping of locale to ... localizations for this command"` would be more appropriate

File: lightbulb/commands/base.py [hunk 5]
```diff
@@ -595,6 +621,16 @@ def app_command_bypass_author_permission_checks(self) -> bool:
...
         return self._initialiser.app_command_bypass_author_permission_checks
 
+    @property
+    def name_localizations(self) -> t.Mapping[t.Union[hikari.Locale, str], str]:
+        """A set of name localizations for this command"""
+        return self._initialiser.name_localizations
+
+    @property
+    def description_localizations(self) -> t.Mapping[t.Union[hikari.Locale, str], str]:
+        """A set of description localizations for this command"""
```
Comment (tandemdude) [inline lightbulb/commands/base.py:631]: Same as above

File: lightbulb/decorators.py [hunk 6]
```diff
@@ -98,13 +98,17 @@ def command(
         inherit_checks (:obj:`bool`): Whether or not the command should inherit checks from the parent group. Only
             affects subcommands. Defaults to ``False``.
         pass_options (:obj:`bool`): Whether the command will have its options passed as keyword arguments when invoked.
+        name_localizations (Mapping[Union[hikari.locales.Locale, :obj:`str`], :obj:`str`]): A set of name localizations for this command. Defaults to an empty dictionary.
```
Comment (tandemdude) [inline lightbulb/decorators.py:101]: Incorrect type in docstring. Should be:
`Mapping[Union[:obj:`hikari.locales.Locale`, :obj:`str`], :obj:`str`]
Change to the above in all places, and change the argument description to something more appropriate as mentioned above.

**Review by tandemdude — CHANGES_REQUESTED** [review]

**Comment by tandemdude** [comment]
You can ignore the mypy failure that is because of hikari dev112

File: lightbulb/commands/base.py (hunk 3 above)
Comment (PerchunPak) [reply lightbulb/commands/base.py:183]: Fixed in 05b489b88d4a2d8dc10a91c1a74c531fe867aacc.

File: lightbulb/commands/base.py (hunk 5 above)
Comment (PerchunPak) [reply lightbulb/commands/base.py]: Fixed in 05b489b88d4a2d8dc10a91c1a74c531fe867aacc.

File: lightbulb/decorators.py (hunk 6 above)
Comment (PerchunPak) [reply lightbulb/decorators.py]: Fixed in 05b489b88d4a2d8dc10a91c1a74c531fe867aacc.

File: lightbulb/commands/base.py (hunk 3 above)
Comment (tandemdude) [reply lightbulb/commands/base.py:183]: You didn't fix these ones

File: lightbulb/commands/base.py [hunk 7]
```diff
@@ -595,6 +621,16 @@ def app_command_bypass_author_permission_checks(self) -> bool:
         """Whether invocations of this command as an application command will bypass author permission checks."""
         return self._initialiser.app_command_bypass_author_permission_checks
 
+    @property
+    def name_localizations(self) -> t.Mapping[t.Union[hikari.Locale, str], str]:
+        """A set mapping of locale to localizations for this command"""
```
[the remark this answers is not in the document]
Comment (PerchunPak) [reply lightbulb/commands/base.py]: Oops, sorry, fixed in 7cb292e5f14c12199eba59f78d08946c2cddfe01.

File: lightbulb/decorators.py [hunk 8]
```diff
@@ -155,6 +159,8 @@ def option(
         autocomplete (Union[:obj:`bool`, AutocompleteCallbackT]): Boolean representing whether the option will use
             autocomplete or the callback to use for autocomplete for this option. This will only affect slash commands.
             Defaults to ``False``.
+        name_localizations (Mapping[Union[:obj:`hikari.locales.Locale`, :obj:`str`], :obj:`str`]): A set of name localizations for this option. Defaults to an empty dictionary.
```
[the remark this answers is not in the document]
Comment (PerchunPak) [reply lightbulb/decorators.py]: Oops, sorry, fixed in 7cb292e5f14c12199eba59f78d08946c2cddfe01.

File: lightbulb/decorators.py [hunk 9]
```diff
@@ -155,6 +159,8 @@ def option(
...
+        description_localizations (Mapping[Union[:obj:`hikari.locales.Locale`, :obj:`str`], :obj:`str`]): A set of description localizations for this option. Defaults to an empty dictionary.
```
[the remark this answers is not in the document]
Comment (PerchunPak) [reply lightbulb/decorators.py]: Oops, sorry, fixed in 7cb292e5f14c12199eba59f78d08946c2cddfe01.

**Pull request merged**